### ISSUE TITLE
InvalidSpecifier raised with installation upgraded with Setuptools 7

### ISSUE THREAD

**Issue by ghost** [pypa/setuptools#263]
Originally reported by: **jaraco (Bitbucket: [jaraco](http://bitbucket.org/jaraco), GitHub: [jaraco](http://github.com/jaraco))**

---

I've installed Setuptools 7.0b1 into my working environment. After doing so, invoking a command from my environment fails with an exception:

```
> vr.cli build Anything 16.8.1
﻿Traceback (most recent call last):
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2315, in _dep_map

  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2390, in __getattr__
    for attr in (
AttributeError: _Distribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "c:\python\scripts\vr.cli.pya", line 5, in <module>
    from pkg_resources import load_entry_point
  File "<frozen importlib._bootstrap>", line 2237, in _find_and_load
  File "<frozen importlib._bootstrap>", line 2226, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 1191, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 1161, in _load_backward_compatible
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2795, in <module>
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 444, in _build_master
    for dist in find_distributions(entry, True):
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 740, in require
    """Remove `dist` from the environment"""
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 640, in resolve
    except ResolutionError:
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2334, in requires
    elif item==bdir and self.precedence==EGG_DIST:
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2329, in _dep_map
    npath= [(p and _normalize_cached(p) or p) for p in path]
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2660, in parse_requirements
    content = []
  File "c:\python\lib\site-packages\setuptools-7.0b1-py3.4.egg\pkg_resources.py", line 2669, in __init__
    def _mkstemp(*args,**kw):
  File "c:\python\lib\site-packages\packaging-14.2-py3.4.egg\packaging\version.py", line 506, in __init__
packaging.version.InvalidSpecifier: Invalid specifier: '>=1.1.5.jaraco.20140924'
```

The vr.cli command comes from the [vr.cli package](https://pypi.python.org/pypi/vr.cli). Prior to [three days ago](https://bitbucket.org/yougov/vr.cli/commits/83a9027c9b613fb62eb97908fd8e48b51bcd36bd?at=default), that project depended on a private release of datadiff to support Python 3. That private release uses setuptools-sanction post release syntax.

With Setuptools 7, that installation of vr.cli 2.1 no longer runs. The setup.py for vr.cli 2.1 also doesn't run. The introduction of this kind of broad incompatibility will probably not be acceptable. I was hoping that the use of PEP440 version specifiers would change very little except sorting. Breaking installed packages can't be acceptable.

If Setuptools is to deprecate these version specifiers, it's going to have to do it incrementally, first by warning of invalid specifiers, then disallowing creation of new projects with those specifiers, then eventually disallowing use of them.

---
- Bitbucket: https://bitbucket.org/pypa/setuptools/issue/263

**Comment by jaraco** [comment]
I believe this was resolved in Nov 2014 with pypa/packaging#15, released as Packaging 14.3.

**Comment by blalterman** [comment]
@jaraco I'm running into a similar problem. How did you fix this?

**Comment by jaraco** [comment]
In pypa/packaging#15, the issue was addressed in packaging 14.3, which was vendored into setuptools. If you're still having issues, it's a new issue, so please make sure you have a recent version of setuptools, and if the issue still occurs there, file a new ticket.
